index.js

```diff
@@ -19,46 +19,168 @@ const getCategory = (score, dementia) => {
 
 	if (
 		washingAndClothing &&
-		(timeAndSpace || dementia) &&
+		timeAndSpace &&
 		continence >= 3 &&
 		(transfers >= 3 || toilet >= 3 || meal >= 3)
 	) {
-		return 'CD';
+		return 'Cd';
 	}
 
 	if (washingAndClothing && transfers >= 3 && toilet >= 3 && (continence >= 3 || meal >= 3)) {
-		if (dementia) {
-			return 'CD';
-		}
-
 		return 'C';
 	}
 
 	if (washingAndClothing && (transfers >= 3 || toilet >= 3 || timeAndSpace)) {
-		if (dementia) {
-			return 'D';
-		}
-
 		return 'B';
 	}
 
-	if (washingOrClothing && !dementia) {
+	if (washingOrClothing) {
 		if (timeAndSpace) {
 			return 'B';
 		}
 
 		return 'A';
 	}
 
-	if (timeAndSpace && !dementia) {
+	if (timeAndSpace) {
 		return 'A';
 	}
 
-	if (dementia) {
-		return 'D';
+	return 'O';
+};
+
+const isIncontinent = score => {
+	const toilet = score.charAt(3);
+
+	return toilet >= 3;
+};
+
+const isIncontinentAndNeedHelpsWithTransferFoodOrToilet = score => {
+	// eslint-disable-next-line no-unused-vars
+	const [washing, clothing, transfers, toilet, continence, meal, time, space] = score;
+
+	return (continence >= 3 && (transfers >= 3 || meal >= 3 || toilet >= 3));
+};
+
+const getCategoryForOBed = (bedType, dementia, palliative) => {
+	if (bedType === 'RVT') {
+		throw new Error('O score cannot be placed in RVT bed');
 	}
 
-	return 'O';
+	if (bedType === 'DVC' && !dementia) {
+		throw new Error('O score cannot be placed in DVC bed if no dementia diagnosis exists');
+	}
+
+	if (bedType === 'DVC' && dementia) {
+		return palliative ? 'Fp' : 'D';
+	}
+
+	// ROB+CVK checks
+	return dementia ? 'D' : 'O';
+};
+
+const getCategoryForABed = (score, bedType, dementia, palliative) => {
+	if (bedType === 'DVC' && !dementia) {
+		throw new Error('A score cannot be placed in DVC bed if no dementia diagnosis exists');
+	}
+
+	if (bedType === 'DVC' && dementia) {
+		return palliative ? 'Fp' : 'D';
+	}
+
+	const isIncontinentAndNeedsExtraHelp = isIncontinentAndNeedHelpsWithTransferFoodOrToilet(score);
+	if (bedType === 'RVT') {
+		if (!dementia || !isIncontinentAndNeedsExtraHelp) {
+			throw new Error('A score cannot be placed in RVT bed if no dementia diagnosis exists or incontinence and additional help');
+		}
+
+		return 'Cd';
+	}
+
+	// ROB+CVK checks
+	if (!dementia) {
+		return 'A';
+	}
+
+	return isIncontinentAndNeedsExtraHelp ? 'Cd' : 'D';
+};
+
+const getCategoryForBBed = (score, bedType, dementia, palliative) => {
+	if (bedType === 'DVC') {
+		if (palliative) {
+			return 'Fp';
+		}
+
+		return dementia ? 'D' : 'F';
```

Comment:
The chart says - when not demented - this can be `F` or `Fd`, but I guess you know what you are doing